src/main/java/com/google/api/codegen/transformer/py/PythonImportSectionTransformer.java

```diff
@@ -0,0 +1,173 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.transformer.py;
+
+import com.google.api.codegen.metacode.InitCodeNode;
+import com.google.api.codegen.transformer.ImportSectionTransformer;
+import com.google.api.codegen.transformer.MethodTransformerContext;
+import com.google.api.codegen.transformer.ModelTypeTable;
+import com.google.api.codegen.transformer.SurfaceTransformerContext;
+import com.google.api.codegen.viewmodel.ImportFileView;
+import com.google.api.codegen.viewmodel.ImportSectionView;
+import com.google.api.codegen.viewmodel.ImportTypeView;
+import com.google.api.tools.framework.model.TypeRef;
+import com.google.common.collect.ImmutableList;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+
+public class PythonImportSectionTransformer implements ImportSectionTransformer {
+  @Override
+  public ImportSectionView generateFileHeaderImportSection(SurfaceTransformerContext context) {
+    // TODO(eoogbe): implement this method when migrating to MVVM
+    return ImportSectionView.newBuilder().build();
+  }
+
+  @Override
+  public ImportSectionView generateInitCodeImportSection(
+      MethodTransformerContext context, Iterable<InitCodeNode> specItemNodes) {
+    ImportSectionView.Builder importSection = ImportSectionView.newBuilder();
+    importSection.appImports(generateInitCodeAppImports(context, specItemNodes));
+    return importSection.build();
+  }
+
+  private List<ImportFileView> generateInitCodeAppImports(
+      MethodTransformerContext context, Iterable<InitCodeNode> specItemNodes) {
+    ImmutableList.Builder<ImportFileView> appImports = ImmutableList.builder();
+    appImports.add(generateApiImport(context));
+    appImports.addAll(generateProtoImports(context, specItemNodes));
+    appImports.addAll(generatePageStreamingImports(context));
+    return appImports.build();
+  }
+
+  private ImportFileView generateApiImport(MethodTransformerContext context) {
+    return createImport(
+        context.getApiConfig().getPackageName(),
+        context.getNamer().getApiWrapperVariableName(context.getInterface()));
+  }
+
+  private List<ImportFileView> generateProtoImports(
+      MethodTransformerContext context, Iterable<InitCodeNode> specItemNodes) {
+    ModelTypeTable typeTable = context.getTypeTable();
+    Set<ImportFileView> protoImports = new TreeSet<>(importFileViewComparator());
+    for (InitCodeNode item : specItemNodes) {
+      TypeRef type = item.getType();
+      if (!type.isRepeated() && type.isMessage()) {
```

Comment:
`foo_pb2` will be imported because of the first line. It's not used in the second. `isRepeated` is excluded because it includes map types that don't end up showing up.